src/com/facebook/buck/cxx/DefaultCxxPlatforms.java

```diff
@@ -42,15 +43,44 @@ private DefaultCxxPlatforms() { }
 
   public static final Flavor FLAVOR = ImmutableFlavor.of("default");
 
-  private static final Path DEFAULT_C_FRONTEND = Paths.get("/usr/bin/gcc");
-  private static final Path DEFAULT_CXX_FRONTEND = Paths.get("/usr/bin/g++");
-  private static final Path DEFAULT_AR = Paths.get("/usr/bin/ar");
-  private static final Path DEFAULT_STRIP = Paths.get("/usr/bin/strip");
-  private static final Path DEFAULT_RANLIB = Paths.get("/usr/bin/ranlib");
-  private static final Path DEFAULT_NM = Paths.get("/usr/bin/nm");
+  private static final Path DEFAULT_C_FRONTEND;
+  private static final Path DEFAULT_CXX_FRONTEND;
+  private static final Path DEFAULT_AR;
+  private static final Path DEFAULT_STRIP;
+  private static final Path DEFAULT_RANLIB;
+  private static final Path DEFAULT_NM;
 
-  private static final Path DEFAULT_OSX_C_FRONTEND = Paths.get("/usr/bin/clang");
-  private static final Path DEFAULT_OSX_CXX_FRONTEND = Paths.get("/usr/bin/clang++");
+  private static final Path DEFAULT_OSX_C_FRONTEND;
+  private static final Path DEFAULT_OSX_CXX_FRONTEND;
+
+  static {
+    ImmutableMap<String, String> env = ImmutableMap.copyOf(System.getenv());
```

Comment:
Instead of using System.getenv() here, please pass it down through the chain of functions all the way from Main.runMainWithExitCode's clientEnvironment (yeah, it's a long way :()

I would possibly also be tempted to just pass down {PATH: "..."} rather than the whole environment map to be explicit about what's being used.